packages/vite/src/node/server/shortcuts.ts

```diff
@@ -0,0 +1,81 @@
+import colors from 'picocolors'
+import type { ViteDevServer } from '..'
+import { openBrowser, resolveBrowserUrl } from './openBrowser'
+
+export function bindShortcuts(server: ViteDevServer): void {
+  if (!server.httpServer) return
+
+  server.config.logger.info(
+    `  > Shortcuts: ` +
+      SHORTCUTS.map((shortcut) => {
+        return `"${shortcut.key}" ${shortcut.name}`
+      }).join(', ')
+  )
+
+  const onInput = (input: string) => {
+    // ctrl+c or ctrl+d
+    if (input === '\x03' || input === '\x04') {
+      return process.kill(process.pid, 'SIGINT')
+    }
+    const shortcut = SHORTCUTS.find((shortcut) => shortcut.key === input)
+    shortcut?.action(server)
+  }
+
+  process.stdin
+    .on('data', onInput)
```

Comment:
Nit: I noticed [vitest listens to the `keypress` event instead of `data`](https://github.com/vitest-dev/vitest/blob/e0dbc30/packages/vitest/src/node/stdin.ts#L79) and [only sets raw mode for TTY](https://github.com/vitest-dev/vitest/blob/e0dbc30/packages/vitest/src/node/stdin.ts#L77-L78). Should we do the same?

Reply:
I'm all in for uniformity!